test/e2e/ipc_uds_test.go

```
@@ -0,0 +1,386 @@
+	select {
+	case r := <-ch:
+		if r.err == io.EOF {
+			return nil, errors.New("stream closed early")
+		}
+		return r.msg, r.err
+	case <-time.After(d):
+		return nil, errors.New("recv timed out")
+	}
+}
+
+func waitFor(t *testing.T, d time.Duration, cond func() bool) {
```

golangci-lint (2.12.2)

[error] func waitFor is unused

(unused)